### ISSUE TITLE
plotProfile() can error if data holds only 1 point

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#2214]
The error below is what I get in a simplified test case.  The problem does not occur in all possible plot types; note that the first plot works fine but the second fails.

``` r
library(oce)
#> Loading required package: gsw
data(ctd)
ctd <- subset(ctd, indices=1)
plotProfile(ctd, ytype = "pressure", xtype = "spiciness0")
```

![](https://i.imgur.com/0caAmK8.png)

``` r
plotProfile(ctd, ytype = "z", xtype = "spiciness0")
#> Error in if (xtype == "index") {: argument is of length zero
```

<sup>Created on 2024-05-13 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by dankelley** [comment]
I know why this happens and will address this today or tomorrow.

**Comment by dankelley** [comment]
This was caused by the automated selection of `ylim` for the plot.  When there is only one point, the formula being used was problematic.  It took me a while to see the problem.  It was because of using `swZ(x[["pressure"]])` in one spot and `object[["z"]]` in another.  The former is not correct, because that will use an assumed latitude, whereas in the latter case, `[[` looks inside the object to determine the latitude.

Anyway, I am pretty sure I fixed the problem (changing a few other things for this odd single-point edge case) a couple of days ago, but forgot to push to GH.  I just pushed it now, in the "develop" branch, commit c99ee3afdd8d20f3a8d909bb7f68bd18e20f831d and so I'm closing this.  I'll reopen it, if after an hour I see that the remote automated checks failed.
